pkg/reconciler/clusteringress/clusteringress.go

```diff
@@ -217,26 +234,43 @@ func (c *Reconciler) reconcile(ctx context.Context, ci *v1alpha1.ClusterIngress)
 	ci.Status.MarkLoadBalancerReady(getLBStatus(gatewayServiceURLFromContext(ctx, ci)))
 	ci.Status.ObservedGeneration = ci.Generation
 
-	// TODO(zhiminx): Istio requires to put certificates under the namespace where Istio ingress
-	// (pods) are deployed so that Istio ingress pods can consume them.
-	// So we need to copy certificates from their origin namespace to the Istio ingress namespace.
-
 	if enablesAutoTLS(ctx) {
+		if !ci.IsPublic() {
+			logger.Infof("ClusterIngress %s is not public. So no need to configure TLS.", ci.Name)
+			return nil
+		}
+
 		// Add the finalizer before adding `Servers` into Gateway so that we can be sure
 		// the `Servers` get cleaned up from Gateway.
 		if err := c.ensureFinalizer(ci); err != nil {
 			return err
 		}
 
-		// TODO(zhiminx): Make Servers based on HTTPProtocol.
-		desiredServers := resources.MakeServers(ci)
-		if err := c.reconcileGateways(ctx, ci, gatewayNames, desiredServers); err != nil {
+		originSecrets, err := resources.GetSecrets(ci, c.secretLister)
+		if err != nil {
+			return err
+		}
+		targetSecrets := resources.MakeSecrets(ctx, originSecrets)
+		if err := c.reconcileCertSecrets(ctx, ci, targetSecrets); err != nil {
 			return err
 		}
+
+		for _, gatewayName := range gatewayNames {
```

Comment:
Same as the above, I think the one step we want to toggle is the gateway reconciliation, but do everything else.